### PR TITLE
[fix bug 1196532] Firefox 42 /whatsnew & /firstrun/learnmore

### PR DESCRIPTION
**Pull request by alexgibson**
URLs:
* `/firefox/42.0/whatsnew/`
* `/firefox/42.0/whatsnew/b/`
* `/firefox/42.0/firstrun/learnmore/`

### CODE REVIEW

**Comment by alexgibson** [comment]
@flodolo - according to the [bug/doc](https://bugzilla.mozilla.org/show_bug.cgi?id=1196532#c2) these strings are final.

**Comment by flodolo** [comment]
Strings extracted locally, please warn if anything changes. I plan to expose them on Friday if I don't hear any objection.

**Comment by alexgibson** [comment]
String change: `Try Private Browsing` has been changed to `Try the new Private Browsing`

**Comment by alexgibson** [comment]
Ok marking this one as ready for review!

**Comment by jpetto** [comment]
@alexgibson - Just a heads up that an additional GTM request has been made in [bug 1217883](https://bugzilla.mozilla.org/show_bug.cgi?id=1217883). (I *think* it's additional anyway. You have similar code, but I think Gareth is asking for something different.)

**Comment by alexgibson** [comment]
Thanks @jpetto - updated

File: media/css/firefox/whatsnew_42/variant-b.less [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    }
+
+    h2 {
+        margin: 0 auto @baseLine * 2;
+        max-width: 15em;
+        .open-sans;
+        .font-size(18px);
+        color: #005189;
+        line-height: 1.3;
+
```
Comment (schalkneethling) [inline media/css/firefox/whatsnew_42/variant-b.less]: Nit: superfluous empty line.

**Comment by schalkneethling** [comment]
Seems that `check-high-res.png` and `check.png` are not used anymore.

File: bedrock/firefox/templates/firefox/whatsnew_42/base.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+{% block body_id %}firefox-whatsnew-42{% endblock %}
+{% block body_class %}{% endblock %}
+
+{% block site_header %}
+<header id="masthead">
+  <div id="tabzilla">
+    <a href="{{ url('mozorg.home') }}">Mozilla</a>
+  </div>
+  {% block up_to_date_message %}
+    <h2>{{ _('Congrats. You’re up to date!') }}</h2>
```
Comment (schalkneethling) [inline bedrock/firefox/templates/firefox/whatsnew_42/base.html]: This message is always shown regardless of the version of Fx I am using. Is that expected?

Comment (schalkneethling) [inline bedrock/firefox/templates/firefox/whatsnew_42/base.html]: There is no logic evident here so, I assume it is. Just wanted to double check to be sure.

File: bedrock/firefox/templates/firefox/whatsnew_42/learnmore.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    {% include 'includes/optimizely.html' %}
+  {% endif %}
+{% endblock %}
+
+{# No need to display up to date messaging on firstrun #}
+{% block up_to_date_message %}{% endblock %}
+
+{% block main_subhead %}
+  {# TODO need to update this link to fall back to the Private Browsing product page #}
+  <a class="button-flat-dark" href="https://support.mozilla.org/kb/private-browsing-use-firefox-without-history/" data-highlight="privateWindow" role="button">{{ _('Try the new Private Browsing') }}</a>
```
Comment (schalkneethling) [inline bedrock/firefox/templates/firefox/whatsnew_42/learnmore.html]: So, should we wait on Private Browsing page landing first or, can these links just be updated in a separate PR?

File: bedrock/firefox/templates/firefox/whatsnew_42/base.html (hunk 2 above)
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew_42/base.html]: This page is only shown to users when they update Firefox, so they will always be "up to date" when they see it, so no need for logic here :+1:

File: bedrock/firefox/templates/firefox/whatsnew_42/learnmore.html (hunk 3 above)
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew_42/learnmore.html]: I think we're good to merge this one and then follow up with another PR once the landing page is merged. I don't think it's really worth holding up?

File: bedrock/firefox/templates/firefox/whatsnew_42/base.html (hunk 2 above)
Comment (schalkneethling) [inline bedrock/firefox/templates/firefox/whatsnew_42/base.html]: Aha! I thought it was something along those lines :+1:

File: bedrock/firefox/templates/firefox/whatsnew_42/learnmore.html (hunk 3 above)
Comment (schalkneethling) [inline bedrock/firefox/templates/firefox/whatsnew_42/learnmore.html]: :+1:

**Comment by alexgibson** [comment]
Thanks for the review @schalkneethling! Added a commit with a few locale tweaks and review updates. I'm going to push this to a demo server to make sure the locale changes look good, but otherwise this should be good for code review.

**Comment by alexgibson** [comment]
Ok, need to make one final l10n tweak, please hold

**Comment by alexgibson** [comment]
Ok locale headline tweaks for `de` and `it` and `es` are now good:

https://www-demo2.allizom.org/de/firefox/42.0/whatsnew/
https://www-demo2.allizom.org/de/firefox/42.0/whatsnew/b/
https://www-demo2.allizom.org/de/firefox/42.0/firstrun/learnmore/

https://www-demo2.allizom.org/it/firefox/42.0/whatsnew/
https://www-demo2.allizom.org/it/firefox/42.0/whatsnew/b/
https://www-demo2.allizom.org/it/firefox/42.0/firstrun/learnmore/

https://www-demo2.allizom.org/es-ES/firefox/42.0/whatsnew/
https://www-demo2.allizom.org/es-ES/firefox/42.0/whatsnew/b/
https://www-demo2.allizom.org/es-ES/firefox/42.0/firstrun/learnmore/

**Comment by schalkneethling** [comment]
r+ :tada:

**Comment by alexgibson** [comment]
@schalkneethling squashed, thanks!
